### DISCUSSION TITLE
Lock fields to prevent changes through directus

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#20996] [Feature Requests]
### Summary

When having fields managed from outside of directus (e.g via code in migrations or external systems) sometimes it's useful to prevent users from making any changes from those fields. This makes sure that the code configuration is the single source of truth.

On non-special fields it's possible by 'missusing' the 'systems' field attribute to at least prevent changes from the admin app:

```ts
filter('fields.read', (items: any) => {
  const protectedUserFields = ['field1', 'field2'];
  items.forEach((item: any, index: number) => {
	  if (item.collection === 'directus_users' && protectedUserFields.includes(item.field)) {
		  items[index].system = true;
	  }
  });
  
  return items;
})
```

For special fields (aliases, groups, etc) this is not possible, as their data are parsed different.

### Motivation

Prevent admins from modifiying single fields / tables that are managed by third party or via code.

### Detailed Design

In directus fields there could be a new column 'locked'. The logic to protect the field could be reused from the 'system' check that already exists.

### Must Have:
- Config to protect field
- Disable changes in UI
- Disable changes from API

### Could Have:
- Option to whitelist some possible changes (e.g allow to change the sort field to reorder the fields)

### Drawbacks

/

### Alternatives

/

### Adoption Strategy

/

**Comment by Dominic-Marcelino** [comment]
Related to:
- #13041
- #17308
- #19800 (Task: "Allow extensions to have an install/uninstall step (for database migrations / other prep)")

**Comment by pedromcipriano** [comment]
It is very useful.
